docs/docs/portals.md

```diff
@@ -0,0 +1,89 @@
+---
+id: portals
+title: Portals
+permalink: docs/portals.html
+---
+
+Portals provide a first-class way to render children into a DOM node that exists outside the DOM hierarchy of the parent component.
+
+```js
+ReactDOM.createPortal(
+  child,
+  container
+);
+```
+
+The first argument (`child`) is any [renderable React child](/docs/react-component.html#render), such as an element, string, or fragment. The second argument (`container`) is a DOM element.
+
+## Usage
+
+Normally, when you return an element from a component's render method, it's mounted into the DOM as a child of the nearest parent node:
+
+```js{4,6}
+render() {
+  // React mounts a new div and renders the children into it
+  return (
+    <div>
+      {this.props.children}
+    </div>
+  );
+}
+```
+
+However, sometimes it's useful to insert a child into a different location in the DOM:
+
+```js{6}
+render() {
+  // React does *not* create a new div. It renders the children into `divNode`.
+  // `divNode` is any valid DOM node, regardless of its location in the DOM.
+  return React.createPortal(
+    this.props.children,
+    divNode,
+  );
+}
+```
+
+A typical use case for portals is when a parent component has an `overflow: hidden` or `z-index` style, but you need the child to visually "break out" of its container.
+
+[Try out an example on CodePen.](https://codepen.io/acdlite/pen/JrKgmz)
+
+## Portals and event bubbling
+
+A nice feature of portals is that, even though the DOM node can be anywhere in the DOM tree, it behaves like a normal React child in every other way. Features like context work exactly the same regardless of whether the child is a portal. 
+
+This includes event bubbling: an event fired from inside a portal will propagate to ancestors in the containing *React tree*, even if those elements are not ancestors in the *DOM tree*:
```

Comment:
Does React `stopPropagation` of the original DOM event itself once it reaches the element controlled by the portal, or does the DOM event continue on its own?

Reply:
No, this only affects how synthetic event propagates.